Code/GraphMol/MarvinParse/MarvinParser.cpp

```diff
@@ -0,0 +1,1110 @@
+            res = new MarvinSuperatomSgroupExpanded(parentMol, molTree);
+          }
+        } else if (role == "SruSgroup" || role == "CopolymerSgroup" ||
+                   role == "ModificationSgroup") {
+          res = new MarvinSruCoModSgroup(parentMol, role, molTree);
+        } else if (role == "MultipleSgroup") {
+          res = new MarvinMultipleSgroup(parentMol, molTree);
+        } else if (role == "MulticenterSgroup") {
+          res = new MarvinMulticenterSgroup(parentMol, molTree);
+        } else if (role == "GenericSgroup") {
+          res = new MarvinGenericSgroup(parentMol, molTree);
+        } else if (role == "MonomerSgroup") {
+          res = new MarvinMonomerSgroup(parentMol, molTree);
+        } else if (role == "DataSgroup") {
+          res = new MarvinDataSgroup(parentMol, molTree);
+        } else {
+          throw FileParseException("Unexpected role " + role + " in MRV file");
+        }
+      }
+
+      BOOST_FOREACH (boost::property_tree::ptree::value_type &v, molTree) {
+        if (v.first == "molecule") {
+          MarvinMolBase *subMol =
+              parseMarvinMolecule(v.second, (MarvinMol *)res);
+          res->sgroups.push_back(std::unique_ptr<MarvinMolBase>(subMol));
+        }
+      }
+
+      return res;
+    } catch (const std::exception &e) {
+      delete res;
+
+      throw;
+    }
+  };
+
+  MarvinReaction *parseMarvinReaction(
+      boost::property_tree::ptree rxnTree,
+      boost::property_tree::ptree documentTree) {
+    auto *res = new MarvinReaction();
+
+    try {
+      boost::property_tree::ptree childTree;
+      bool foundChild = false;
+      try {
+        childTree = rxnTree.get_child("reactantList");
+        foundChild = true;
+      } catch (const std::exception &e) {
+        foundChild = false;
+      }
+
+      if (foundChild) {
+        BOOST_FOREACH (boost::property_tree::ptree::value_type &v, childTree)
+          res->reactants.push_back(std::move(std::unique_ptr<MarvinMol>(
+              (MarvinMol *)parseMarvinMolecule(v.second))));
+      }
+
+      try {
+        childTree = rxnTree.get_child("agentList");
+        foundChild = true;
+      } catch (const std::exception &e) {
+        foundChild = false;
+      }
+      if (foundChild) {
+        BOOST_FOREACH (boost::property_tree::ptree::value_type &v, childTree)
+          res->agents.push_back(std::move(std::unique_ptr<MarvinMol>(
+              (MarvinMol *)parseMarvinMolecule(v.second))));
+      }
+
+      try {
+        childTree = rxnTree.get_child("productList");
+        foundChild = true;
+      } catch (const std::exception &e) {
+        foundChild = false;
+      }
+      if (foundChild) {
+        BOOST_FOREACH (boost::property_tree::ptree::value_type &v, childTree)
+          res->products.push_back(std::move(std::unique_ptr<MarvinMol>(
+              (MarvinMol *)parseMarvinMolecule(v.second))));
+      }
+
+      // <arrow type="DEFAULT" x1="-11.816189911577812" y1="-10.001443743444021"
```

Comment:
From here down you are doing a fair amount of parsing work to pull out information which is never used when constructing an RDKit molecule.
I don't think it makes sense to delete it since you've done the work and we might want to do something with it in the future, but I also don't think we should be paying the price to do the extra parsing by default.
So I'd suggest either surrounding this with and `#ifdef` which causes it to not be built by default, or adding an extra argument, defaulting to false, which controls whether or not this info is parsed.